os/logm/logm.h

```diff
@@ -45,12 +73,24 @@ extern "C" {
 #define EXTERN extern
 #endif
 
-EXTERN volatile int g_logm_isready;
+EXTERN volatile int g_logm_head;
+EXTERN volatile int g_logm_tail;
+EXTERN volatile int g_logm_available;
+EXTERN volatile int g_logm_enqueued_count;
+EXTERN volatile int g_logm_overflow_offset;
+EXTERN volatile int g_logm_dropmsg_count;
+EXTERN char *volatile g_logm_rsvbuf;
+EXTERN volatile int logm_bufsize;
```

Comment:
This need not be volatile and can be removed from this header file. All set values need be volatile and could be removed from header file.